forch/authenticator.py

```diff
@@ -73,12 +76,37 @@ def do_mab_request(self, src_mac, port_id):
 
     def process_device_placement(self, src_mac, device_placement):
         """Process device placement info and initiate mab query"""
+        LOGGER.info('Anurag process_device_placement %s', src_mac)
         if not device_placement.connected:
             LOGGER.warning("Device not connected. Ignoring auth request for %s", src_mac)
             return
         portid_hash = ((device_placement.switch + str(device_placement.port)).encode('utf-8')).hex()
         port_id = int(portid_hash[:6], 16)
-        self.do_mab_request(src_mac, port_id)
+        #self.do_mab_request(src_mac, port_id)
+        if src_mac not in self.sessions:
+            self.sessions[src_mac] = MacAuthBypassStateMachine(
+                src_mac, port_id, self.radius_query.send_mab_request, self.process_session_result)
```

Comment:
likely better to just pass in self.radius_query -- but that's a small difference so not necessary to change